app/src/main/java/unipd/se18/ocrcamera/JSONReportParser.java

```diff
@@ -0,0 +1,116 @@
+package unipd.se18.ocrcamera;
+
+import android.util.Log;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import static android.support.constraint.Constraints.TAG;
+
+//All methods are static so no need for an dedicated class, i made it to have a clearer code while im writing it
+//The methods have to be added to the class that calls them
+//The class refers to TestEntry class
+//The class can receive only a particular type of JSONObject to have a idea see the example at this link:
+// https://ingsw18.slack.com/archives/CDFU418Q4/p1541678187026900
+
+public class JSONReportParser {
+
+
+    /**
+     * Convert a string in JSONObject and call metod JSONReader
+     * @param fileContent a String in JSON format
+     * @return TestEntry[] an array of Test Entry with all the JSONObject value inside or null if fileContent is not in JSON format
+     * @author Giovanni Furlan (gr2)
+     */
+    public static TestEntry[] parseReport(String fileContent)
+    {
+        TestEntry[] array = null;
+        try
+        {
+            JSONObject json = new JSONObject(fileContent);
+            array = retrieveEntries(json);
+        }
+        catch(JSONException e)
+        {
+            Log.e(TAG, "Error in parseReport");
+        }
+        return array;
+    }
+
+
+
+    /**
+     * Get a JSONObject with keys linked to JSONObjects A that contains the informations.
+     * Put informations of a JSONObject A in a TestEntry, do it for all JSONObject and return all the TestEntry in an array
+     * @param json JSONObject file extracted by PhotoTester class
+     * @return TestEntry[] an array of Test Entry with all the JSONObject inside
+     * @author Giovanni Furlan (gr2) - modified by Leonardo Rossi (g2)
+     */
+    private static TestEntry[] retrieveEntries(JSONObject json)
+    {
+
+        TestEntry[] entryArray = new TestEntry[json.length()];
+        try
+        {
+            //Iterator that slides all JSONObject inside the param
+            Iterator<String> keys = json.keys();
+            int i = 0;
+            while(keys.hasNext())
+            {
+                //Add all the information inside a TestEntry
+                JSONObject jsonDetails = json.getJSONObject(keys.next());
+                TestEntry entry = extractValues(jsonDetails);
+                entryArray[i]=entry;
+                i++;
+            }
+        }
+        catch(JSONException e)
+        {
+            Log.e(TAG, "Error in JSONReader");
+        }
+        return entryArray;
+    }
+
+    /**
+     * Add all the given JSONObject value in a TestEntry
+     * @param json the JSONObject where to find the value
+     * @return TestEntry with all the JSONObject value
```

Comment:
what happens if the json object doesn't have the fields "original_name" or "confidence"or  "ingredients" or "notes"?

Reply:
the method don't work, i have to add that on the specifics, because the JSONObject given have always those fields

Reply:
I think the method should manage those cases better because the programmer who use the method cannot guarantee that the json object have all the fields "original_name","confidence", "ingredients", "notes" and "tags".  If the json have a missing field then all the rest of TestEntry would be thrown away. Maybe you could put a try catch block each time you use json.getString and manage those cases separately. This is just my opinion.

Reply:
Thanks for the advice but I think that this way the code would be not clear and difficult to read

Reply:
ok